services/sqs/src/main/java/software/amazon/awssdk/services/sqs/batchmanager/BatchOverrideConfiguration.java

```diff
@@ -0,0 +1,449 @@
+            this.maxBufferSize = maxBufferSize;
+            return this;
+        }
+
+        /**
+         * Define the maximum amount of time that an outgoing call waits for other requests before sending out a
+         * batch request.
+         * TODO : Decide if Ms needs to be added to the name in surface API review meeting
+         * @param maxBatchOpenDuration The new maxBatchOpenDuration value.
+         * @return This object for method chaining.
+         */
+        public Builder maxBatchOpenDuration(Duration maxBatchOpenDuration) {
+            this.maxBatchOpenDuration = maxBatchOpenDuration;
+            return this;
+        }
+
+        /**
+         * Define the custom visibility timeout to use when retrieving messages from SQS. If set to a value greater than zero,
+         * this timeout will override the default visibility timeout set on the SQS queue. Set it to -1 to use the default
+         * visibility timeout of the queue. Visibility timeout of 0 seconds is not supported.
+         *
+         * @param visibilityTimeout The new visibilityTimeout value.
+         * @return This object for method chaining.
+         */
+        public Builder visibilityTimeout(Duration visibilityTimeout) {
+            this.visibilityTimeout = visibilityTimeout;
+            return this;
+        }
+
+        /**
+         * Define the amount of time, the receive call will block on the server waiting for messages to arrive if the
+         * queue is empty when the receive call is first made. This setting has no effect if long polling is disabled.
+         *
+         * @param longPollWaitTimeout The new longPollWaitTimeout value.
+         * @return This object for method chaining.
+         */
+        public Builder longPollWaitTimeout(Duration longPollWaitTimeout) {
+            this.longPollWaitTimeout = longPollWaitTimeout;
+            return this;
+        }
+
+        /**
+         * Define the minimum wait time for incoming receive message requests. Without a non-zero minimum wait time, threads can
+         * easily waste CPU time busy-waiting against empty local buffers. Avoid setting this to 0 unless you are confident
+         * threads will do useful work in-between each call to receive messages!
+         *
+         * @param minReceiveWaitTime The new minReceiveWaitTime value.
+         * @return This object for method chaining.
+         */
+        public Builder minReceiveWaitTime(Duration minReceiveWaitTime) {
+            this.minReceiveWaitTime = minReceiveWaitTime;
+            return this;
+        }
+
+        /**
+         * Define the maximum number of concurrent receive message batches. The greater this number, the faster the queue will be
+         * pulling messages from the SQS servers (at the expense of consuming more threads).
+         *
+         * @param maxInflightReceiveBatches The new maxInflightReceiveBatches value.
+         * @return This object for method chaining.
```

Comment:
There's a lot of knobs here, and I don't fully understand how they relate. Is there a summary somewhere that someone can read to understand how the threads, queues, etc. work together?